tests_integration/testing_utils/snowpark_services_utils.py

```diff
@@ -103,30 +97,51 @@ def drop_service(self, service_name: str) -> None:
                 "object",
                 "drop",
                 "service",
-                f"{self.database}.{self.schema}.{service_name}",
+                self._get_fqn(service_name),
             ],
         )
         assert result.json[0] == {  # type: ignore
             "status": f"{service_name.upper()} successfully dropped."
         }
 
-    def wait_until_service_will_be_finish(self, service_name: str) -> None:
-        wait_counter = 0
-        max_counter = 90
-        while wait_counter < max_counter:
+    def wait_until_service_is_ready(self, service_name: str) -> None:
+        self._wait_until_service_reaches_state(service_name, "READY", 900)
+
+    def _wait_until_service_reaches_state(
+        self, service_name: str, target_status: Union[str, dict], max_duration: int
+    ):
+        assert max_duration > 0
+        max_counter = math.ceil(max_duration / 10)
+        if isinstance(target_status, str):
+            target_status = {"status": target_status}
+        for i in range(max_counter):
             status = self._execute_status(service_name)
-            if contains_row_with(
-                status.json,
-                {
-                    "serviceName": service_name.upper(),
-                    "status": "DONE",
-                    "message": "Completed successfully",
-                },
-            ):
+            if contains_row_with(status.json, target_status):
                 return
+            elif contains_row_with(status.json, {"status": "FAILED"}):
```

Comment:
We do see some transient failures happening in service creation. The service reached transient FAILED status at first time. Then K8s retried and the service was READY again. Could we keep waiting for READY status until timeout, not fail right way when getting FAILED status(this may make test flaky)?

Reply:
sounds good, will make this change